src/ILCompiler.ReadyToRun/src/Compiler/DependencyAnalysis/ReadyToRun/ExceptionInfoLookupTableNode.cs

```diff
@@ -0,0 +1,182 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using ILCompiler.DependencyAnalysisFramework;
+using Internal.Text;
+using Internal.TypeSystem;
+
+using Debug = System.Diagnostics.Debug;
+
+namespace ILCompiler.DependencyAnalysis.ReadyToRun
+{
+    class SyntheticSymbolNode : ISymbolDefinitionNode
```

Comment:
What is SyntheticSymbolNode? I apologize but this name seems to be very generic to me.

Reply:
Well, SyntheticSymbolNode is just what it says on the tin - in the exception info lookup table, we need to emit relocs against each of the EH info blobs for the individual methods. The way EH info gets produced in CorInfoImpl i.e. in the form of an ObjectData, makes it somewhat challenging to obtain the symbols representing these for the purpose of relocating the lookup table - this is why I introduced these replacement "synthetic nodes" to represent them.

Frankly speaking, I think we should ultimately have a somewhat deeper code split between R2R and RyuJit here as ObjectData is sort of the worst option especially as with my changes the EH info blobs are simple 24-byte blocks without any relocs. This would let us drop the "nested ObjectData" logic in the object writer and likely get rid of the synthetic node too.

Reply:
As I said in @nattress's #6322 - if the way how we publish EH/debug/GC/unwind info for the fully AOT case is getting in the way, we should just split the publishing logic off into partial classes and make something convenient for the readytorun case. All of the `PublishCode`, `EncodeEHInfo` etc should just be in the AOT-specific file and we can invent whatever is convenient for the ReadyToRun writer.

That's why I wanted ReadyToRun to not derive from `MethodCodeNode` and eventually I would like to delete the whole IMethodCodeNode business too (David introduced this for the RyuJIT prototype, but we should have just started doing a partial class for CorInfoImpl at that point) - https://github.com/dotnet/corert/pull/6098#discussion_r203272897. E.g. it pains me we have `RhEHClauseKind` in the common CorInfoImpl even though it's Redhawk specific.

Reply:
I think GC and unwind info is fine (at least for now) as they seem to be using the same format as CoreCLR. The EH info has popped up exactly because its format differs between the runtimes. I'm going to investigate how to simplify this as I don't deny getting rid of the synthetic symbol node and of the nested ObjectData logic sounds compelling.